streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskExecutor.java

```diff
@@ -96,6 +96,7 @@ private int processTask(final Task task, final int maxNumRecords, final long beg
         try {
             while (processed < maxNumRecords && task.process(now)) {
                 task.clearTaskTimeout();
+                task.setConsumedOffsetsAndMetadata(taskManager.offsetAndMetadataPerTask().get(task));
```

Comment:
It seems we are updating the consumedoffsetsAndMetadata in every single iteration of the process loop. But it seems, they change only once after every poll.

It seems we are already passing the `ConsumerRecords` object into `addRecordsToTasks`. Couldn't we call `task.setConsumedOffsetsAndMetadata` once inside `addRecordsToTasks` (for all tasks)?

Reply:
good point but adding records to tasks does not mean that they have been processed.